### PR TITLE
check for duplicate keys in Java properties files, fix CodeMeta displayName

### LINKED ISSUE #9169: Duplicate property key in Bundle.properties file V5.12.1
Hi Dataverse Community,

We detected a duplicate keys in the Bundle.properties file : 

advanced.search.datasets.persistentId.tip=The persistent identifier for the Dataset.
advanced.search.datasets.persistentId.tip=The Dataset's unique persistent identifier, either a DOI or Handle

best regards
Steven Ferey.

### PR DESCRIPTION
**Pull request by poikilotherm**
**What this PR does / why we need it**:
Fix duplicated keys in all `*.properties`. Add Github Action to check for no re-introduction of dups.

**Which issue(s) this PR closes**:

Closes #9169

**Special notes for your reviewer**:
Feel free to push to this branch for fixing all of the dups in the same go. I don't mind!

**Suggestions on how to test this**:
Look at the PRs Action log

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:
Nope

**Is there a release notes update needed for this change?**:
Nope

**Additional documentation**:
None

### CODE REVIEW

**Comment by poikilotherm** [comment]
This PR now contains a test commit (33731ad), which needs to be reverted before merging.

Here is the output of the verification job:
image

Here is the output of the duplication check job:
image

That one also leaves notes in the changes tab:
image

**Comment by poikilotherm** [comment]
As you can see, it already detected a missing property in CodeMeta and a typo - fixed with 1aeb665.

File: tests/check_duplicate_properties.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    # Scan the whole file for duplicates
+    FILTER=$(grep -a -v -E "^(#.*|\s*$)" "$FILE" | cut -d"=" -f1 | sort | uniq -c | tr -s " " | { grep -vs "^ 1 " || true; })
+
+    # If there are any duplicates present, analyse further to point people to the source
+    if [ -n "$FILTER" ]; then
+        FAIL=1
+
+        echo "::group::$FILE"
+        for KEY in $(echo "$FILTER" | cut -d" " -f3); do
```
Comment (pdurbin) [inline tests/check_duplicate_properties.sh:20]: On Mac I get `cut: stdin: Illegal byte sequence` when it reaches lines like `datasetfieldtype.titulo.title=T?tulo`

There's a long discussion at https://stackoverflow.com/questions/19242275/re-error-illegal-byte-sequence-on-mac-os-x

I dunno, I'm ok with this only working on Linux and in our CI.

File: src/main/java/propertyFiles/codeMeta20.properties [hunk 2]
```diff
@@ -1,5 +1,6 @@
 metadatablock.name=codeMeta20
-metadatablock.displayName=Software Metadata (CodeMeta 2.0)
+metadatablock.displayName=Software Metadata (CodeMeta v2.0)
```
Comment (pdurbin) [inline src/main/java/propertyFiles/codeMeta20.properties:2]: This is a good fix, making it the same as what's in scripts/api/data/metadatablocks/codemeta.tsv

**Review by pdurbin — APPROVED** [review]
I can't run this on my Mac, but that's ok. I ran the check_duplicate_properties.sh script on Linux and it worked fine.

I didn't install jbang and Graal to run the other script. We only need these to work in CI.

Approved.

**Comment by pdurbin** [comment]
We're having a wee problem with this script. See the [Slack thread](https://iqss.slack.com/archives/C03R1E7T4KA/p1725631227524999) and this issue:

- #10826

**Pull request merged**